.github/maintainers_guide.md

```diff
@@ -3,102 +3,146 @@
+several packages, and lerna makes it easier to manage operations across the set of packages. The very next thing you
+should do is run `npx lerna bootstrap` in the root of this project to install dependencies for each of the packages,
+and to allow lerna to link them to each other.
 
 ## Tasks
 
 ### Testing
 
-This package has unit tests for most modules (files) in the `test` directory. The structure mirrors the `lib` directory closely. You can run the entire test suite using the npm script `npm test`. This command is also executed by Travis, the continuous integration service, for every Pull Request and branch. The coverage is computed with the `istanbul` package. The tests themselves are run using the `mocha` test runner.
+This package has tests for individual packages in the respective `test` directory inside those packages. It also has
+integration tests in the `integration-tests` directory at the root. You can run the entire test suite using the npm
+script `npm test` at the top level. This script will use Lerna to invoke tests in each package and the integration
+tests.
+
+Tests are executed by Travis, our continuous integration system, slightly differently. Travis runs several, more
+granular builds in order to report on success and failure in a more useful (targetted) way. There is one build for each
+package on each supported version of Node, as well as one for the integration tests on each supported version of Node.
+Travis also runs linting in each package, which is separate from tests so that you can run tests locally frequently
+without having to block for fixing styling problems. Lastly, Travis uploads the coverage report for the tests ran within
+the build to Codecov, our coverage reporting system. Travis reports status on each PR. Codecov aggregates all the
+coverage reports, and separate reports status on each PR. The configuration is stored in `.travis.yml`.
 
-Test code should be written in syntax that runs on the oldest supported Node.js version, without transpiling. This ensures that backwards compatibility is tested and the APIs look reasonable in versions of Node.js that do not support the most modern syntax.
+Test code should be written in syntax that runs on the oldest supported Node.js version, without transpiling. This
+ensures that backwards compatibility is tested and the APIs look reasonable in versions of Node.js that do not support
+the most modern syntax.
 
-A useful trick for debugging inside tests is to use the Chrome Debugging Protocol feature of Node.js to set breakpoints and interactively debug. In order to do this you must run mocha directly. This means that you should have already linted the source (`npm run lint`), manually. You then run the tests using the following command: `./node_modules/.bin/mocha test/{test-name}.js --debug-brk --inspect` (replace {test-name} with an actual test file).
+**TODO** update the following information for debugging with VSCode or with the protocol directly.
```

Comment:
Was this meant to be left in?

Reply:
Reading further, it seems like it is since there are other TODOs. Disregard if so!

Reply:
yep! far from perfect (otherwise i would have closed #748 with this), but i think leaving them in and getting the bulk of the useful info out is better than waiting for this to be complete.